### ISSUE TITLE
Add `MultipleChoice` example with Keras Core and Keras NLP

### ISSUE THREAD

**Issue by awsaf49** [keras-team/keras-core#791]
`keras.io/examples` offers an impressive collection of NLP examples, but it doesn't have any examples on a multiple-choice task. I've recently created a Kaggle notebook for the [**LLM Science Exam**](https://www.kaggle.com/competitions/kaggle-llm-science-exam) competition. The notebook demonstrates how to approach this task using Keras, Core, and KerasNLP libraries also it focuses on building a  building model from scratch using a typical NLP classifier.

**Notebook**: https://www.kaggle.com/code/awsaf49/llm-science-exam-kerascore-kerasnlp-tpu

I'm open to suggestions on adapting it for inclusion in `keras.io/examples`.

**Comment by awsaf49** [comment]
cc: @fchollet

**Comment by fchollet** [comment]
Thanks for the suggestion -- I think that would be a valuable addition to Keras.io/examples. Does it work with JAX too? Is there an open dataset for the task, or would we have to use the dataset from Kaggle?

**Comment by awsaf49** [comment]
> Does it work with JAX

I think that is the best part, it works with all the backends even with PyTorch. But PyTorch is a bit slower as may be it doesn't have `jit_compile` option in `model.fit()`.  For tensorflow it consumes all the RAM as mentioned in https://github.com/keras-team/keras-core/pull/473

> Is there an open dataset for the task, or would we have to use the dataset from Kaggle?

We can use the [**SWAG**](https://rowanzellers.com/swag/) (EMNLP'18) dataset which is available on [GitHub](https://github.com/rowanz/swagaf/tree/master/data). 

Just out of curiosity, you mentioned **kaggle-dataset** is not open is this because it requires **Accepting Rules of Competition** which ultimately needs **Kaggle Account** ?

**Comment by fchollet** [comment]
> But PyTorch is a bit slower as may be it doesn't have jit_compile

Yes, this is generally true in most cases. We'll add compilation support to the PyTorch backend in the future (it doesn't work out of the box). But even then, it will be difficult to match the performance of JAX/XLA.

> For tensorflow it consumes all the RAM as mentioned in https://github.com/keras-team/keras-core/pull/473

Did the `TF_FORCE_GPU_ALLOW_GROWTH` fix the issue or not?

> We can use the [SWAG](https://rowanzellers.com/swag/) (EMNLP'18) dataset which is available on [GitHub](https://github.com/rowanz/swagaf/tree/master/data).

Sounds good!

> Just out of curiosity, you mentioned kaggle-dataset is not open, is this because it requires Accepting Rules of Competition which ultimately needs Kaggle Account or because kaggle-dataset in general can't be downloaded using wget call thus must use kaggle-api?

Both -- these are significant hurdles in a code example. First, you need to create a Kaggle account and verify it. Then, you need to accept the terms of that specific competition. Then, you need to create a Kaggle API key and download it. Finally, you need to upload your key to your Colab notebook (a step which itself has like 3 substeps). It's just a lot of friction, and it prevents any form of automation -- it requires a human in the loop.

**Comment by awsaf49** [comment]
Anyway, I'll start working on the PR as it is already working for `jax` and `torch`. I will also check if the problem with `tensorflow` is specifically for `kaggle` or not. 

I genuinely appreciate your time, especially considering your current health situation with the fever. Wishing you a swift recovery. Thank you.
